app/models/enterprise.rb

```diff
@@ -332,6 +332,10 @@ def can_invoice?
     abn.present?
   end
 
+  def instagram=(value)
+    write_attribute(:instagram, value.try(:gsub, %r{\Ahttps?://(?:www.)?instagram.com/([a-zA-Z0-9._]{1,30})/?\z}, '@\1'))
```

Comment:
Dots `.` have a special meaning in regular expressions and match any character. We should consider that. I think this would be right:
```ruby
regex = %r{\Ahttps?://(?:www\.)?instagram\.com/([a-zA-Z0-9._]{1,30})/?\z}
write_attribute(:instagram, value.try(:gsub, regex, '@\1'))
```
I also extracted the expression into a variable to simplify it and comply with our line length restrictions.